client/consensus/slots/src/lib.rs

```diff
@@ -583,9 +597,87 @@ pub fn slot_lenience_linear(parent_slot: u64, slot_info: &SlotInfo) -> Option<Du
 	}
 }
 
+/// Trait for providing the strategy for when to backoff block authoring.
+pub trait BackoffAuthoringBlocksStrategy<N> {
+	/// Returns true if we should backoff authoring new blocks.
+	fn should_backoff(
+		&self,
+		chain_head_number: N,
+		chain_head_slot: u64,
+		finalized_number: N,
+		slow_now: u64,
+	) -> bool;
+}
+
+/// A simple default strategy for how to decide backing off authoring blocks if the number of
+/// unfinalized blocks grows too large.
+#[derive(Clone)]
+pub struct SimpleBackoffAuthoringBlocksStrategy<N> {
```

Comment:
```suggestion
pub struct BackOffBlockAuthoringOnFinalizedHeadLagging<N> {
```

Something like this? Maybe not that good, but the other name also didn't really fit :P